internal/go_repository_test.go

```diff
@@ -44,6 +44,20 @@ go_repository(
     version = "v0.8.1",
     sum ="h1:iURUrRGxPUNPdy5/HRSm+Yj6okJ6UtLINN0Q9M4+h3I=",
 )
+
+go_repository(
+    name = "com_github_apache_mesos",
+    build_file_proto_mode = "package",
+    directives = [
+        "gazelle:proto_strip_import_prefix /include/",
+        "gazelle:exclude include/mesos/attributes.hpp",
+        "gazelle:exclude include/mesos/executor.hpp",
+        "gazelle:exclude include/mesos/hook.hpp",
+    ],
+    importpath = "github.com/apache/mesos",
+    sum = "h1:65atl5ZUZ7fzMM/oIVFV5VdO0plKO0hye9taZR6ssHw=",
+    version = "v0.0.0-20190710134757-4ae064484664",
+)
```

Comment:
I don't think the test will actually build this yet. Add a new test function `TestDirectives` that builds a target that wouldn't otherwise build.

If you can avoid using protos in the test, that would be ideal. protoc takes a long, long time to compile if it's not cached.

The simplest way I can think of would be to `# gazelle:exclude` some directory, then run `bazel query @repo//...` and verify that that rules were not generated for that directory.